docs/openai.md

```diff
@@ -0,0 +1,592 @@
+query uses [jsonb operators](https://www.postgresql.org/docs/current/functions-json.html#FUNCTIONS-JSON-PROCESSING)
+and a [jsonpath query](https://www.postgresql.org/docs/current/functions-json.html#FUNCTIONS-SQLJSON-PATH)
+to process [the response](https://platform.openai.com/docs/api-reference/moderations/object)
+from OpenAI:
+
+```sql
+create table commit_history_moderated 
+( id int not null primary key
+, detail text -- the content that was moderated
+, flagged_categories jsonb -- an array of json strings
+);
+
+insert into commit_history_moderated (id, detail, flagged_categories)
+select
+  x.id
+, x.detail
+  -- pull out the list of only the categories that were flagged
+, jsonb_path_query_array(x.moderation, '$.results[0].categories.keyvalue() ? (@.value == true).key')
+from
+(
+    select
+      id
+    , detail
+      -- call the openai api using the pgai extension. the result is jsonb
+    , openai_moderate('text-moderation-stable', detail) as moderation
+    from commit_history
+) x
+where (x.moderation->'results'->0->>'flagged')::bool -- only the ones that were flagged
+;
+```
+
+### Summarization
+
+Use the pgai extension to summarize content. In a single query, ask for a summarization
+of a month's worth of git commits in the form of release notes in Markdown format. You
+provide one message for the system and another one for the user.
+
+The git commits for the month are appended in text format to the user message. This query
+uses jsonb operators to pull out the content of the [response](https://platform.openai.com/docs/api-reference/chat/object) only:
+
+```sql
+-- the following two metacommands cause the raw query results to be printed
+-- without any decoration
+\pset tuples_only on
+\pset format unaligned
+
+-- summarize and categorize git commits to produce a release notes document
+select openai_chat_complete
+( 'gpt-4o'
+, jsonb_build_array
+  ( jsonb_build_object
+    ( 'role', 'system'
+    , 'content', 'You are a software release engineer who summarizes git commits to produce release notes.'
+    )
+  , jsonb_build_object
+    ( 'role', 'user'
+    , 'content'
+    , -- build up a list of the commit details to append to the prompt
+      concat
+      ( E'Summarize the following list of commits from the timescaledb git repo from August 2023 in a release notes document in markdown format.\n\n'
+      , string_agg(x.commit_desc, E'\n\n')
+      )
+    )
+  )
+)->'choices'->0->'message'->>'content'
+from
+(
+    -- convert each to a text format
+    select format
+    ( E'%s %s\n\tcommit: %s\n\tauthor: %s\n\tdate: %s\n\tdetail: %s'
+    , row_number() over (order by "date")
+    , summary
+    , "commit"
+    , author
+    , "date"
+    , detail
+    ) as commit_desc
+    from commit_history
+    -- just look at commits from August 2023
+    where date_trunc('month', "date") = '2023-08-01 00:00:00+00'::timestamptz
+    order by "date"
+) x
+;
+```
```

Comment:
I was thinking about we also put a link to the raw examples in the examples folder in the end as an encouragement to let people pick the scripts and run :)


> If you're interested in run this or  other full examples, check the [examples](../examples) folder for more.